docs/cli/migration/list-nullability.md

```diff
@@ -0,0 +1,192 @@
+---
+title: Amplify Codegen Models - List and list components nullability
+description: Generating Amplify models with Amplify CLI x.y.z
+---
+
+### **What is changing?** 
+
+Amplify CLI x.y.z has updated the codegen process to correctly set the optionality for types in models.
+
+_Schema example_
+
+```
+type ListStringContainer @model {
+  stringList: [String!]!
+  stringNullableList: [String!]
+  nullableStringList: [String]!
+  nullableStringNullableList: [String]
+}
+```
```

Comment:
Nit: it would be easier to keep track of which thing you're talking about if you maintained parallelism between the case names. Genericizing the type to "Element" in the name also makes it a bit clearer what you're referring to:

requiredElementRequiredList: [String!]!
optionalElementOptionalList: [String]
requiredElementOptionalList: [String!]
optionalElementRequiredList: [String]!